### PR TITLE
initial commit to handle new hdw.dat format

### PR DESCRIPTION
**Pull request by kevinkrieger**
both year and sc not used in backscatter
Not yet tested

### CODE REVIEW

File: backscatter/core/core.py [hunk 1]
```diff
@@ -1,52 +1,70 @@
...
-                            hdw_dict["phasesign"] = float(fields[11])
-                            hdw_dict["interoffx"] = float(fields[12])
-                            hdw_dict["interoffy"] = float(fields[13])
-                            hdw_dict["interoffz"] = float(fields[14])
-                            hdw_dict["rxrisetime"] = float(fields[15])
-                            hdw_dict["attenstages"] = float(fields[16])
-                            hdw_dict["maxgates"] = int(fields[17])
-                            hdw_dict["maxbeams"] = int(fields[18])
-
-                            hdw_info[hdw_dict["stid"]] = hdw_dict
```
Comment (RemingtonRohel) [inline backscatter/core/core.py:48]: I'm gonna wager a guess that this line needs to be kept around

Comment (kevinkrieger) [reply backscatter/core/core.py:48]: I will not take that wager haha

Comment (RemingtonRohel) [reply backscatter/core/core.py:48]: Upon reading this with fresh eyes, I'm not sure what I was thinking

**Comment by RemingtonRohel** [comment]
With these new commits, I can now install and use backscatter. I tested processing a small rawacf file to fitacf with the new command-line syntax `python -m backscatter.fitacf infile outfile`, which seemed to work successfully. 

For the larger discussion:
* I have hardcoded in using `tdiff_a`, but this is not a long-term viable solution. We need to find a way to handle this intelligently.
* Will changing the command-line syntax break code elsewhere? We may need to ask around about this.

**Comment by RemingtonRohel** [comment]
I've added support for `tdiff` as a specifiable option when fitting data. If not specified, the default is to use `tdiff_a` from the hdw file. The full usage help message is now:

```
usage: python -m backscatter.fitacf [-h] [--tdiff TDIFF] [--debug-mode] infile outfile

positional arguments:
  infile         Input rawacf file path
  outfile        Output file path

optional arguments:
  -h, --help     show this help message and exit
  --tdiff TDIFF  Tdiff value to use for processing. If not specified, tdiff_a from hdw file is used. Units of us
  --debug-mode   Disables multiprocessing usage to produce more meaningful exception at the cost of performance.
```

File: backscatter/fitacf/fitacf.py [hunk 2]
```diff
@@ -459,43 +461,50 @@ def _fit(raw_data, debug_mode=False):
...
-        fitted_records = [_fit(rr,True) for rr in raw_records]
+        fitted_records = [_fit(rr, args.tdiff, True) for rr in raw_records]
     else:
-        fitted_records = fit(raw_records)
+        fitted_records = fit(raw_records, args.tdiff)
 
     dm.dicts_to_file(fitted_records,args.outfile,file_type='fitacf')
+
+
+if __name__ == "__main__":
```
Comment (kevinkrieger) [inline backscatter/fitacf/fitacf.py:506]: I suppose we can search our usages of backscatter to see where we might be calling it like this.

File: backscatter/fitacf/determinations.py [hunk 3]
```diff
@@ -15,15 +15,21 @@
...
     of the plasma
 
     """
 
-    def __init__(self,raw_data,range_list,noise_pwr):
+    def __init__(self,raw_data,range_list,noise_pwr, tdiff):
         hdw_info = hdw[raw_data['stid']]
+
+        if tdiff is None:
+            self.tdiff = hdw_info['tdiff_a']        # TODO: Can we incorporate tdiff_b somehow?
```
Comment (kevinkrieger) [inline backscatter/fitacf/determinations.py:29]: Good question. I think this is tricky as we would need to know which radars use it - and then parse the filename for 'a' or 'b' in the channel designator. Maybe...

**Review by kevinkrieger — COMMENTED** [review]
Looks good. I can't approve as the author, but I approve.

**Comment by RemingtonRohel** [comment]
Going to merge this PR and kick the can down the road with the whole tdiff channel issue. I think that for the regular use case of this repo (realtime conversion with Borealis radars) this will be a mostly non-issue.

**Pull request merged**